grpcbp/prometheus_test.go

```diff
@@ -0,0 +1,57 @@
+package grpcbp
+
+import (
+	"testing"
+)
+
+func TestServiceAndMethodSlug(t *testing.T) {
+	testCases := []struct {
+		name        string
+		fullMethod  string
+		wantService string
+		wantMethod  string
+	}{
+		{
+			name:        "success",
+			fullMethod:  "/package.service/method",
+			wantService: "package.service",
+			wantMethod:  "method",
+		},
+		{
+			name:        "success 2",
+			fullMethod:  "package.service/method",
+			wantService: "package.service",
+			wantMethod:  "method",
+		},
+		{
+			name:        "extra /",
+			fullMethod:  "/foo/bar/baz",
+			wantService: "foo",
+			wantMethod:  "bar/baz",
+		},
+		{
+			name:        "no /",
+			fullMethod:  "package.service.method",
+			wantService: "",
+			wantMethod:  "",
+		},
+		{
+			name:        "empty input",
+			fullMethod:  "",
+			wantService: "",
+			wantMethod:  "",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			gotService, gotMethod := serviceAndMethodSlug(tt.fullMethod)
+			if got, want := gotService, tt.wantService; got != want {
+				t.Errorf("got %v, want %v", got, want)
```

Comment:
if the output was
```
got "a", want "b"
got "c", want "d"
```

you wouldn't know which was which.  So we prefix them:

```suggestion
				t.Errorf("serviceAndMethodSlug(%q).service = %v, want %v",  tt.fullMethod, got, want)
```

Thus, if you see this output

```
serviceAndMethodSlug("/package.service/method").service = "", want "package.service"
serviceAndMethodSlug("/package.service/method").method = "package.service", want "method"
```

you actally don't need to look at the test at all, it includes all the info you need to understand the failure.